units/src/amount.rs

```diff
@@ -231,6 +224,50 @@ impl std::error::Error for ParseDenominationError {
     }
 }
 
+/// Parsing error, unknown denomination.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct UnknownDenominationError(InputString);
+
+impl fmt::Display for UnknownDenominationError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "unknown denomination {}", self.0.display_cannot_parse("bitcoin denomination"))
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for UnknownDenominationError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> { None }
+}
+
+impl From<&str> for UnknownDenominationError {
+    fn from(s: &str) -> Self { Self(s.into()) }
+}
+
+/// Parsing error, possibly confusing denomination.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct PossiblyConfusingDenominationError(InputString);
+
+impl fmt::Display for PossiblyConfusingDenominationError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "possibly confusing denomination {}. We intentionally do not support 'M' and 'P' so as to not confuse mega/milli and peta/pico",
+            self.0.display_cannot_parse("bitcoin denomination")
```

Comment:
There's no nice API in `InputString` to make the message exactly this but maybe this will be fine:

```rust
write!(f, "{}: "possibly confusing denomination - we intentionally do not support 'M' and 'P' so as to not confuse mega/milli and peta/pico", self.0.display_cannot_parse("bitcoin denomination"))
```

Reply:
Used suggestion (with quotes fixed).